test/test_autograd.py

```diff
@@ -6188,6 +6192,258 @@ def foo(a):
+
+            gfoo, gbar = torch.autograd.grad(p.sum(), (foo, bar), retain_graph=True, allow_unused=True)
+            self.assertEqual(gfoo, torch.ones_like(foo))
+            self.assertIsNone(gbar)
+
+            gfoo, gbar = torch.autograd.grad(t.sum(), (foo, bar), retain_graph=True, allow_unused=True)
+            self.assertIsNone(gfoo)
+            self.assertEqual(gbar, torch.ones_like(bar))
+
+            # Check that forward gradients are not impacted by detach
+            detached_dual = dual.detach()
+            out = detached_dual * 2
+            p, t = fwAD.unpack_dual(out)
+            self.assertFalse(p.requires_grad)
+            self.assertFalse(t.requires_grad)
+            self.assertEqual(p, foo * 2)
+            self.assertEqual(t, bar * 2)
+
+            # Check that forward gradients are not impacted by no_grad
+            with torch.no_grad():
+                out = dual * 3
+            p, t = fwAD.unpack_dual(out)
+            self.assertFalse(p.requires_grad)
+            self.assertFalse(t.requires_grad)
+            self.assertEqual(p, foo * 3)
+            self.assertEqual(t, bar * 3)
+
+            # Check that forward gradients are not impacted by inplace detach
+            dual = dual.clone()
+            dual.detach_()
+            out = dual * 2
+            p, t = fwAD.unpack_dual(out)
+            self.assertFalse(p.requires_grad)
+            self.assertFalse(t.requires_grad)
+            self.assertEqual(p, foo * 2)
+            self.assertEqual(t, bar * 2)
+
+    def test_view_detach(self):
+        foo = torch.rand(1)
+        bar = torch.ones(1)
+        baz = torch.rand(2)
+
+        with fwAD.dual_level():
+            dual = fwAD.make_dual(foo, bar)
+
+            baz_detach = baz.detach()
+            baz_view = baz_detach.narrow(0, 0, 1)
+            baz_view *= dual
+
+            self.assertIsNotNone(fwAD.unpack_dual(baz)[1])
+            self.assertIsNotNone(fwAD.unpack_dual(baz_detach)[1])
+            self.assertIsNotNone(fwAD.unpack_dual(baz_view)[1])
+
+    def test_view_inplace(self):
+        foo = torch.rand(2)
+        bar = torch.ones(2)
+
+        with fwAD.dual_level():
+            # Note that in this test, we use "update" to mean computing the right tangent for the dual
+            # All the inplace operations here are expected to update the primal value of the Tensors but
+            # not always their tangents.
+
+            # Check that inplace ops do not update non-differentiable views
+            cloned_foo = foo.clone()
+            cloned_bar = bar.clone()
+            # Non differentiable view
+            dual = fwAD.make_dual(cloned_foo, cloned_bar)
+            dual *= 2
+            # Check that non differentiable view was not updated
+            self.assertIsNone(fwAD.unpack_dual(foo)[1])
+            # Check that the computed result is correct
+            self.assertEqual(cloned_bar, bar * 2)
+            self.assertEqual(fwAD.unpack_dual(dual)[1], bar * 2)
```

Comment:
`self.assertEqual(fwAD.tangent(), bar * 2)`

Reply:
At the very least make `unpack_dual` return a named tuple so you don't need to positionally index it. (BTW if you make it an OP and name the returns, this will happen automatically).

Reply:
The problem with that is that it would dump it in `torch.` but yes I am considering it.